packages/remix-react/components.tsx

```diff
@@ -20,6 +20,7 @@ import {
   useResolvedPath,
 } from "react-router-dom";
 import type { LinkProps, NavLinkProps } from "react-router-dom";
+import type { LoaderFunction } from "@remix-run/server-runtime";
```

Comment:
This makes `@remix-run/server-runtime` a dependency of `@remix-run/react`, which is currently not the case.

Reply:
It's not a runtime dependency since `import type` gets stripped out entirely at build time. Put differently, this works regardless of whether `@remix-run/server-runtime` is in the `package.json` of `remix-react`. That said, I only did this avoid avoid a redundant definition of `LoaderFunction` but it's easy enough to duplicate it - is that preferred? I wasn't sure what the preferred pattern was for sharing types across package boundaries.

Reply:
`LoaderFunction` is used in a type that's exported (`DataOrLoader`, which is used in exported `UseLoaderData` type), so that makes `@remix-run/server-runtime` a dependency.

Reply:
I've updated the PR to remove this import.